src/services/dom/errors.ts

```diff
@@ -70,3 +73,26 @@ export class ServerError extends Error {
     this.status = status;
   }
 }
+
+export class DocumentGetElementByIdError extends ErrorService.HvBaseError {
+  name = 'DocumentGetElementByIdError';
+
+  constructor(id: string, doc: Document, error: Error) {
+    super(
+      `Document.getElementById failed for id: ${id} on doc: ${docToString(
+        doc,
+      )} and error: ${error.message}`,
+    );
+    this.stack = error.stack;
+    this.setExtraContext('error', error);
+  }
+}
+
+const docToString = (doc: Document): string => {
+  try {
+    const serializer = new XMLSerializer();
+    return serializer.serializeToString(doc);
+  } catch {
+    return 'serializing error';
```

Comment:
it's probably worth getting the error message
```suggestion
  } catch (e) {
    return `serializing error: ${e.message}`;
```